crates/wasmparser/src/validator/component.rs

```diff
@@ -808,26 +818,7 @@ impl ComponentState {
                         .add_import(import, types, offset)?;
                 }
                 crate::ComponentTypeDeclaration::Alias(alias) => {
-                    match alias {
-                        crate::ComponentAlias::Outer { kind, count, index }
-                            if kind == ComponentOuterAliasKind::CoreType
-                                || kind == ComponentOuterAliasKind::Type =>
-                        {
-                            match kind {
-                                ComponentOuterAliasKind::CoreType => {
-                                    Self::alias_core_type(components, count, index, types, offset)?
-                                }
-                                ComponentOuterAliasKind::Type => {
-                                    Self::alias_type(components, count, index, types, offset)?
-                                }
-                                _ => unreachable!(),
-                            }
-                        }
-                        _ => return Err(BinaryReaderError::new(
-                            "only outer type aliases are allowed in component type declarations",
-                            offset,
-                        )),
-                    }
+                    Self::add_alias(components, alias, types, offset)?;
```

Comment:
This removes the better error message of "only outer type aliases are allowed..." for component/instance type declarations in favor of a more confusing error message of "unknown instance X: instance index out of bounds".

The latter makes it seem like the encoding was legal except the index was wrong when instead instance export aliasing is not legal in type declarations.

I'm of the opinion that if we're going to allow illegal encoding of aliases in type definitions in `wasm-encoder`, we should do the best validation we can against said encodings in `wasmparser`.